backend/src/data/repositories/course/course.repository.ts

```diff
@@ -90,6 +90,21 @@ class Course {
       .castTo<CourseGetResponseDto>()
       .execute();
   }
+
+  public async updateCategory(
+    courseId: number,
+    newCategoryId: number,
+  ): Promise<CourseGetResponseDto> {
+    return this.#CourseModel
+      .query()
+      .patchAndFetchById(courseId, {
+        courseCategoryId: newCategoryId,
+      })
+      .withGraphJoined('vendor')
+      .first()
```

Comment:
As far as I know, `patchAndFetchById` should return one element, I think there's no need for `.first()`